shell/common/node_includes.h

```diff
@@ -30,11 +30,32 @@
 #include "tracing/agent.h"
 
 #include "electron/pop_node_defines.h"
-
-// Alternative to NODE_MODULE_CONTEXT_AWARE_X.
-// Allows to explicitly register builtin modules instead of using
-// __attribute__((constructor)).
-#define NODE_LINKED_MODULE_CONTEXT_AWARE(modname, regfunc) \
-  NODE_MODULE_CONTEXT_AWARE_CPP(modname, regfunc, nullptr, NM_F_LINKED)
+#include "shell/common/gin_helper/locker.h"
+
+#define ELECTRON_MODULE(modname, regfunc, cond)                \
+  static node::node_module _module = {NODE_MODULE_VERSION,     \
+                                      NM_F_LINKED,             \
+                                      nullptr,                 \
+                                      __FILE__,                \
+                                      nullptr,                 \
+                                      regfunc,                 \
+                                      NODE_STRINGIFY(modname), \
+                                      nullptr,                 \
+                                      nullptr};                \
+  void _register_##modname() {                                 \
+    if (cond)                                                  \
+      node_module_register(&_module);                          \
+  }
+
+#define ELECTRON_BROWSER_MODULE(modname, regfunc)      \
+  ELECTRON_MODULE(electron_browser_##modname, regfunc, \
+                  gin_helper::Locker::IsBrowserProcess())
```

Comment:
I suggest to move the process based registration of these modules to https://github.com/electron/electron/blob/3f4c4a4470d7eb089f53f947bd40f3eac4cc6c3f/shell/common/node_bindings.cc#L352-L365 and base it on `kProcessType` value instead which is more consistent.

Reply:
done